### DISCUSSION TITLE
Feature: Environments

### DISCUSSION THREAD

**Discussion by iamclaytonray** [supabase/supabase#9801] [General]
# Feature request

## Is your feature request related to a problem? Please describe.

Currently, there is no way to switch between dev, staging, & prod (for my use case, I'm sure different envs for different people).

## Describe the solution you'd like

I'd love to hit different URIs for different envs. I doubt I need to go into more detail but when I ship a feature, I want it to go to staging before prod. QA can do their thing but not affect prod data, etc. When it's time, I'd like to transition staging to prod. 

## Describe alternatives you've considered

I haven't tried it but I suppose I could manually create different projects, manually replicating the database, etc. This isn't scalable or efficient though

## Additional context

I feel like this is pretty self-explanatory but if I need to add more details, I can. I think the simplest example would be Heroku but really, most any major platforms have some variant of this

**Comment by Mr-K9** [comment]
hxjjx

**Comment by kiwicopple** [comment]
Nice one @iamclaytonray - we'll release database branching at some point, just need to knock over a few easier tasks first.

In the meantime, you're right - the way to do it is to have 2 projects.

**Reply by MildTomato** [reply]
[Database branching discussion](https://github.com/orgs/supabase/discussions/18937)
Early access to Branching is currently in progress.

**Comment by iamclaytonray** [comment]
100%! I'd rather have functions before this and don't consider this feature critical (but not minor either). Just wanted to add visibility to add to the roadmap 🙂

**Comment by sampl** [comment]
Related: https://github.com/supabase/supabase/discussions/542

**Comment by kiwicopple** [comment]
We also have a document describing how to do this with github actions:

https://supabase.com/docs/guides/cli/managing-environments
